Neighborly/VectorList.cs

```diff
@@ -552,29 +390,24 @@ public bool Update(Guid Id, Vector vector)
 
         lock (_lock)
         {
-            int index = FindIndexById(Id);
+            int index = _vectorList.IndexOf(vector);
+            var updated = false;
             if (index != -1)
             {
-                if (_vectorList[index] != null)
-                {
-                    _vectorList[index] = vector;
-                }
-                else if (_onDiskFilePaths[index] != string.Empty)
-                {
-                    SaveToDisk(vector, _onDiskFilePaths[index]);
-                }
-                else
-                {
-                    // This should never happen
-                    throw new FileNotFoundException();
-                }
+                _vectorList[index] = vector;
+                updated = true;
             }
             else
             {
-                return false;
+                updated = _memoryMappedList.Update(vector);
```

Comment:
The test case for this `Update` method is failing after the rebase, because the `Contains` check only checks for equality based on the `Vector.Id`.

Should `Id`, `OriginalText` or `Tags` count for equality and the `Contains` check, or is the equality contract just `a.Values == b.Values`? This PR's implementation of a memory mapped list is solely based on searching by the `Vector.Id`, so that other definitions might be slow to search/remove. Also, if anything other than `Id` is relevant for the `Contains` check, should a call `Add` do anything if the `Values` already exist? (That check might be quite slow 😅) That would kinda make it work a bit more like an `ISet<Vector>` than an `ICollection<Vector>` though?